internal/databases/postgres/daemon_handler.go

```diff
@@ -0,0 +1,176 @@
+package postgres
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path"
+
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type SocketMessageType byte
+
+const (
+	CheckType    SocketMessageType = 'C'
+	FileNameType SocketMessageType = 'F'
+)
+
+type SocketMessageHandler interface {
+	Handle(messageBody []byte) error
+}
+
+type ArchiveMessageHandler struct {
+	messageType SocketMessageType
+	fd          net.Conn
+	uploader    *WalUploader
+}
+
+func (h *CheckMessageHandler) Handle(messageBody []byte) error {
+	_, err := h.fd.Write([]byte{'O'})
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Error on writing in socket: %v \n", err)
+		return err
+	}
+	tracelog.InfoLogger.Println("Successful configuration check")
+	return nil
+}
+
+type CheckMessageHandler struct {
+	messageType SocketMessageType
+	fd          net.Conn
+	uploader    *WalUploader
+}
+
+func (h *ArchiveMessageHandler) Handle(messageBody []byte) error {
+	err := messageValidation(messageBody)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Incorrect message: %v\n", err)
+		return err
+	}
+	tracelog.InfoLogger.Printf("wal file name: %s\n", string(messageBody))
+	PgDataSettingString, ok := internal.GetSetting(internal.PgDataSetting)
+	if !ok {
+		tracelog.ErrorLogger.Print("\nPGDATA is not set in the conf.\n")
+		return fmt.Errorf("PGDATA is not set in the conf")
+	}
+	pathToWal := path.Join(PgDataSettingString, "pg_wal")
+	fullPath := path.Join(pathToWal, string(messageBody))
+	tracelog.InfoLogger.Printf("starting wal-push for %s\n", fullPath)
+	err = HandleWALPush(h.uploader, fullPath)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Failed to archive file: %s, err: %v \n", string(messageBody), err)
+		return err
+	}
+	_, err = h.fd.Write([]byte{'O'})
+	if err != nil {
+		tracelog.ErrorLogger.Printf("failed to write in socket: %v\n", err)
+		return err
+	}
+	return nil
+}
+
+func messageValidation(messageBody []byte) error {
+	if len(messageBody) < 24 {
```

Comment:
Also, we archive not only WALs but other files with length other than 24.

I think we can remove this validation.